File: fieldservice_mobile/lib/models/order.dart
import 'treatment.dart';

class Order {
  final int id;
  final String customerName;
  final String customerPhone;
  final String? contactPhone;
  final String address;
  final double lat;
  final double lng;
  final int treatmentId;
  final String? scope;
  final int? technicianId;
  final String scheduledDate; // "2026-04-15"
  final String scheduledStart; // "09:00:00"
  final String scheduledEnd; // "10:00:00"
  final String status; // draft, assigned, in_progress, completed
  final String? paymentMethod;
  final double price;
  final String? notes;
  final Treatment? treatment;

  Order({
    required this.id,
    required this.customerName,
    required this.customerPhone,
    this.contactPhone,
    required this.address,
    required this.lat,
    required this.lng,
    required this.treatmentId,
    this.scope,
    this.technicianId,
    required this.scheduledDate,
    required this.scheduledStart,
    required this.scheduledEnd,
    required this.status,
    this.paymentMethod,
    required this.price,
    this.notes,
    this.treatment,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      customerName: json['customerName'] ?? '',
      customerPhone: json['customerPhone'] ?? '',
      contactPhone: json['contactPhone'],
      address: json['address'] ?? '',
      lat: (json['lat'] ?? 0).toDouble(),
      lng: (json['lng'] ?? 0).toDouble(),
      treatmentId: json['treatmentId'] ?? 0,
      scope: json['scope'],
      technicianId: json['technicianId'],
      scheduledDate: json['scheduledDate'] ?? '',
      scheduledStart: json['scheduledStart'] ?? '',
      scheduledEnd: json['scheduledEnd'] ?? '',
      status: json['status'] ?? 'draft',
      paymentMethod: json['paymentMethod'],
      price: (json['price'] ?? 0).toDouble(),
      notes: json['notes'],
      treatment: json['treatment'] != null
          ? Treatment.fromJson(json['treatment'])
          : null,
    );
  }

  String get startTimeShort => scheduledStart.substring(0, 5);
  String get endTimeShort => scheduledEnd.substring(0, 5);
  bool get isCompleted => status == 'completed';
  bool get isInProgress => status == 'in_progress';
}

File: FieldService/Models/Order.cs
namespace FieldService.Models;

/// <summary>
/// Zlecenie — centralna tabela systemu.
/// Dane klienta wpisywane bezpośrednio przez handlowca (nie z bazy klientów).
/// </summary>
public class Order
{
    public int Id { get; set; }
    
    // ---- Dane klienta (wpisywane przez handlowca) ----
    
    /// <summary>Imię i nazwisko osoby zamawiającej</summary>
    public string CustomerName { get; set; } = string.Empty;
    
    /// <summary>Numer telefonu zamawiającego</summary>
    public string CustomerPhone { get; set; } = string.Empty;
    
    /// <summary>Telefon kontaktowy na miejscu — jeśli inny niż zamawiającego</summary>
    public string? ContactPhone { get; set; }
    
    /// <summary>Adres zabiegu</summary>
    public string Address { get; set; } = string.Empty;
    
    /// <summary>Współrzędne adresu — do obliczania odległości techników</summary>
    public double Lat { get; set; }
    public double Lng { get; set; }
    
    // ---- Zabieg ----
    
    /// <summary>Rodzaj zabiegu — wybrany z listy (dropdown)</summary>
    public int TreatmentId { get; set; }
    public Treatment Treatment { get; set; } = null!;
    
    /// <summary>
    /// Zakres zabiegu — tekst wpisany przez handlowca.
    /// Np. "Mieszkanie 3-pokojowe, 65m², kuchnia i łazienka"
    /// </summary>
    public string? Scope { get; set; }
    
    // ---- Przypisanie technika ----
    
    public int? TechnicianId { get; set; }
    public Technician? Technician { get; set; }
    
    public int CreatedByUserId { get; set; }
    public User CreatedBy { get; set; } = null!;
    
    // ---- Czas ----
    
    public DateOnly ScheduledDate { get; set; }
    public TimeOnly ScheduledStart { get; set; }
    public TimeOnly ScheduledEnd { get; set; }
    
    // ---- Status i szczegóły ----
    
    public string Status { get; set; } = "draft";
    public string? PaymentMethod { get; set; }
    public decimal Price { get; set; }
    public string? Notes { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Nawigacja
    public Protocol? Protocol { get; set; }
    public Payment? Payment { get; set; }
}
